contracts/modules/cover/CoverUtilsLib.sol

```diff
@@ -160,4 +162,36 @@ library CoverUtilsLib {
     // cast last 20 bytes of hash to address
     return IStakingPool(address(uint160(uint(hash))));
   }
+
+  function performStakeBurn(
+    uint coverId,
+    uint burnAmount,
+    ICoverNFT coverNFT,
+    CoverData storage cover,
+    CoverSegment memory segment,
+    PoolAllocation[] storage allocations,
+    bytes32 stakingPoolProxyCodeHash,
+    uint globalCapacityRatio
+  ) external returns (address /* owner */) {
+
+    // increase amountPaidOut only *after* you read the segment
+    cover.amountPaidOut += SafeUintCast.toUint96(burnAmount);
+
+    uint allocationCount = allocations.length;
+    for (uint i = 0; i < allocationCount; i++) {
+
+      PoolAllocation memory allocation = allocations[i];
+
+      uint nxmBurned = allocation.coverAmountInNXM
```

Comment:
Would `burnAmountInNXM` be more in line with `payoutAmountInNXM`?

Reply:
nxmBurned is how much its burned because you apply all the factors, think this is ok

Reply:
ah i see what you mean

Reply:
we did a rename for payoutAmountInNXM

There is no variable called burnAmountInNXM

Maybe you want to rename burnAmount?